File: android/src/main/java/com/github/quadflask/react/navermap/ClickableRNNaverMapFeature.java
package com.github.quadflask.react.navermap;

import android.content.Context;

import androidx.annotation.NonNull;

import com.naver.maps.map.overlay.Overlay;

public class ClickableRNNaverMapFeature<T extends Overlay> extends RNNaverMapFeature<T> implements Overlay.OnClickListener {
    public ClickableRNNaverMapFeature(EventEmittable emitter, Context context) {
        super(emitter, context);
    }

    @Override
    public void addToMap(RNNaverMapView map) {
        super.addToMap(map);
        feature.setOnClickListener(this);
    }

    @Override
    public boolean onClick(@NonNull Overlay overlay) {
        super.onClick(overlay);
        return false;
    }
}

File: android/src/main/java/com/github/quadflask/react/navermap/ClickableRNNaverMapFeature.kt
package com.github.quadflask.react.navermap

import android.content.Context
import com.naver.maps.map.NaverMap
import com.naver.maps.map.overlay.Overlay

open class ClickableRNNaverMapFeature<T : Overlay>(
  emitter: EventEmittable,
  context: Context
) : RNNaverMapFeature<T>(emitter, context), Overlay.OnClickListener {
  override fun addToMap(map: NaverMap) {
    super.addToMap(map)
    feature.onClickListener = this
  }

  override fun onClick(overlay: Overlay): Boolean {
    emitEvent("onClick", null)
    return true
  }
}
